teuthology/dispatcher/supervisor.py

```diff
@@ -0,0 +1,202 @@
+                                     teuth_config.results_timeout)
+            args.extend(['--timeout', str(timeout)])
+            if job_config.get('email'):
+                args.extend(['--email', job_config['email']])
+        # Execute teuthology-results, passing 'preexec_fn=os.setpgrp' to
+        # make sure that it will continue to run if this worker process
+        # dies (e.g. because of a restart)
+        result_proc = subprocess.Popen(args=args, preexec_fn=os.setpgrp)
+        log.info("teuthology-results PID: %s", result_proc.pid)
+        return
+
+    log.info('Creating archive dir %s', job_config['archive_path'])
+    safepath.makedirs('/', job_config['archive_path'])
+    log.info('Running job %s', job_config['job_id'])
+
+    suite_path = job_config['suite_path']
+    arg = [
+        os.path.join(teuth_bin_path, 'teuthology'),
+    ]
+    # The following is for compatibility with older schedulers, from before we
+    # started merging the contents of job_config['config'] into job_config
+    # itself.
+    if 'config' in job_config:
+        inner_config = job_config.pop('config')
+        if not isinstance(inner_config, dict):
+            log.warn("run_job: job_config['config'] isn't a dict, it's a %s",
+                     str(type(inner_config)))
+        else:
+            job_config.update(inner_config)
+
+    if verbose or job_config['verbose']:
+        arg.append('-v')
+
+    arg.extend([
+        '--owner', job_config['owner'],
+        '--archive', job_config['archive_path'],
+        '--name', job_config['name'],
+    ])
+    if job_config['description'] is not None:
+        arg.extend(['--description', job_config['description']])
+    arg.append('--')
+
+    with tempfile.NamedTemporaryFile(prefix='teuthology-worker.',
+                                     suffix='.tmp', mode='w+t') as tmp:
+        yaml.safe_dump(data=job_config, stream=tmp)
+        tmp.flush()
+        arg.append(tmp.name)
+        env = os.environ.copy()
+        python_path = env.get('PYTHONPATH', '')
+        python_path = ':'.join([suite_path, python_path]).strip(':')
+        env['PYTHONPATH'] = python_path
+        log.debug("Running: %s" % ' '.join(arg))
+        p = subprocess.Popen(args=arg, env=env)
+        log.info("Job archive: %s", job_config['archive_path'])
+        log.info("Job PID: %s", str(p.pid))
+
+        if teuth_config.results_server:
+            log.info("Running with watchdog")
+            try:
+                run_with_watchdog(p, job_config)
+            except Exception:
+                log.exception("run_with_watchdog had an unhandled exception")
+                raise
+        else:
+            log.info("Running without watchdog")
+            # This sleep() is to give the child time to start up and create the
+            # archive dir.
+            time.sleep(5)
+            symlink_worker_log(job_config['worker_log'],
+                               job_config['archive_path'])
```

Comment:
There is no sense in this code since we can redirect output of corresponding dispatcher log directly to archive_path.

Reply:
Removed this symlink since the supervisor log is already inside the job archive now.